clusters/app.ci/hive/README.md

```diff
@@ -43,4 +43,131 @@ hiveadmission-9f7df866b-zb98l       1/1     Running   0          2m16s
 ```
 
 ## Deploy a cluster via Hive
-TBU
+
+### SSH key
+Managed by `ci-secret-bootstrap`: `jenkins-ci-iam-ssh-key`.
+
+### Pull Secret
+
+Create a secret for pull-secret which will be used by `ClusterDeployment` later.
+
+```bash
+### reuse the existing pull secret, we use this pull-secret to generate the install-config above.
+$ oc --context build01 extract secret/pull-secret -n openshift-config
```

Comment:
Isn't this secret on all clusters?

Reply:
Yes. I copy the content to a secret in `hive` on `app.ci`.
We could use build01's or build02's pull-secret, not on `app.ci`. It is not managed by us.

Reply:
True but we create test clusters all the time for which we need a secret, why not use that instead of re-using the one from openshift-config?

Reply:
I am confused.
Which secret do you want me to use?

Reply:
Let me try with another perspective.
The created secret is `test01-pull-secret` in hive project.
It is referred in the clusterdeployment later.

```yaml
apiVersion: hive.openshift.io/v1
kind: ClusterDeployment
metadata:
  name: test01
  namespace: hive
spec:
  baseDomain: hive.aws.ci.openshift.org
  clusterName: test01
  platform:
    aws:
      credentialsSecretRef:
        name: jenkins-ci-iam-aws-creds
      region: us-east-1
  provisioning:
    releaseImage: quay.io/openshift-release-dev/ocp-release:4.6.0-x86_64
    installConfigSecretRef:
      name: test01-install-config
    sshPrivateKeySecretRef:
      name: jenkins-ci-iam-ssh-key
  pullSecretRef:
    name: test01-pull-secret 
```

The pull secret has to be in the same namespace as the ClusterDeployment.
I put everything in hive because it is simple for getting it to run for the first time.

In production, I think (not yet tested) the `ClusterDeployment` will be in `ci-op-xxxx` namespace.
we could manage one pull-secret (say `global-hive-pull-secret`) in hive namespace.
And then specify it in `HiveConfig`

```yaml
apiVersion: hive.openshift.io/v1
kind: HiveConfig
metadata:
  name: hive
spec:
  globalPullSecretRef:
    name: global-pull-secret
 logLevel: debug
 targetNamespace: hive
```

This way we do not need to set up a pull secret in each CR of clusterdeployment.
https://github.com/openshift/hive/blob/master/docs/using-hive.md#pull-secret

I did not start it with the global one because the example in the doc uses a local one.
It is not managed by our secret tool because it is only for demo purpose.

Or without setting up the global one,  `ci-operator` copies the secret from hive namespace to `ci-op-xxxx` namespace like image distributor. And all other referred resources, like `test01-install-config` etc.

Anyway, that would be https://issues.redhat.com/browse/DPTP-2045.
In the current card, all we need to show is that we can use hive to provision a cluster, right?

Reply:
What I meant is: When we install clusters during CI runs, we have a pull secret for them. We should re-use that very pull secret for hive, instead of abusing the one from one of the build clusters.

Reply:
I see. Probably this is the one for the current CI runs to install a cluster:

https://github.com/openshift/release/blob/fc05d4d777a2bd9d1733ae18e626ac112434c8b8/core-services/ci-secret-bootstrap/_config.yaml#L835-L879

That one is currently not available on `app.ci`.

So back to you original question
> Isn't this secret on all clusters?

The answer should have been no. :)
All clusters except `api.ci`.

This readme is only for demoing that hive works.
We can decide which pull-secret to use for CI later, right?

Reply:
Is there anything blocking the usage of a pull secret that is not an actual production one?

Reply:
Updated.